### DISCUSSION TITLE
Use the asset store component for project properties

### DISCUSSION THREAD

**Discussion by arthuro555** [4ian/GDevelop#2456] [Enhancement ideas (and their technical discussions)]
## Description
Some extensions may require to add extension properties, but, for me at least, I am always hesitant to do so in fear of bloating the project properties editor, and often use a "setup action" instead (an action you call once in the game with some parameters, for example broker server connection on p2p). If the properties could be showed by extensions, this would help add lots of properties that are useful but would otherwise bloat the list (for example all electron initial window parameter instead of having to use the actions in the first scene).

## Mockup
![image](https://user-images.githubusercontent.com/19349038/111876957-4e388900-89a1-11eb-9ffc-012b8bdb3c3f.png)

**Comment by 4ian** [comment]
So we have a UX problem here that would be indeed interesting to tackle.
When we have "too much" things, we should think:
- can we split them? (for example, tabs? accordions?).
  - we could have properties grouped by type: game properties, monetization, online services.
- can we set up a search (not sure if it make sense here, project properties are not a "store")
- can we think about another navigation pattern? 
  - Here you suggest filters, which is a way to think about it, but again we don't really have a store. I think we should prefer another pattern.
  - We could think about a Tree view (like in IntelliJ preferences)

(Note that the components in AssetStore/ folder are, well, for "stores" of things: resources, extensions, assets - which comes from the GDevelop apis. Here we are in front of a UX/UI issue, where we should use a different approach).

We should again further: 
why is it a problem that they are too much things? It's because it will bloat this dialog. Why is this a problem? Because users will see a lot of things they don't care about, and will lose track of things they will care about.
So, should we:
- Make links from actions/conditions to this property dialog when it's useful? Like if you use a P2P action/condition, there should be a link "Click here to set up the default server properties". When clicked, the game properties dialog open. That's an idea!
- Don't show everything by default. Just the basic stuff. Then when we can have buttons "Set up AdMob", "Set up P2P", "Set up Firebase". When clicked, this could reveal the fields related to this (and maybe a link to some help page). We could also reveal the fields automatically if they are not empty, or if an action/condition of the extension is used in the game?

**Reply by 4ian** [reply]
As an example, we could investigate the usage of vertical tabs :) 
![image](https://user-images.githubusercontent.com/1280130/111877982-da48b180-899d-11eb-93a5-ecf2f5841855.png)

**Comment by Bouh** [comment]
project properties, and for preferences, keyboard shortcuts, hints too.
